forwards_paper.tex

```diff
@@ -816,80 +816,134 @@ \subsubsection*{Sequential simplification and prior history}
-Suppose instead that we have run a simulation of $2N$ individuals for $T$ generations,
-begun with no prior history.
-One way to answer this question is to ask:
-How many of the resulting $4NT$ edges are required after simplification?
-As above, the expected number of edges is bounded by $2N-2$ plus four times the mean marginal tree length.
-Roughly, the length of time for which a marginal tree has $k$ tips is
-$2N/(k(k-1)) = 2N(1/(k-1) - 1/k)$ generations,
-and so the time to go from $N$ to $n$ tips is $2N(1/(n-1) - 1/(N-1))$.
-This implies that after running for time $T$ we expect to
-have around $n(T)$ roots, where $n(T) \approx 1 + 2N/(T+2)$.
+What if $T$ is smaller:
+how many of the resulting $2NT$ edges are required after simplification?
+In other words, how fast does the information in the pedigree become irrelevant?
+% \citet{padhukasahasram2008exploring} obtained impressive speedups relative to contemporary simulators
+% by retaining only 8 generations of history.
+% This begs the question: how fast does the information in the pedigree become irrelevant?
+Now, we need to compute the expected total length of all branches in a coalescent tree
+up until time $T$ (or the common ancestor, whichever comes first).
+Again, coalescent theory tells us that
+the expected length of time for which a coalescent tree has $k$ lineages
+is $2N/(k(k-1)) = 2N(1/(k-1) - 1/k)$ generations.
+Since the tree has $k$ branches over this period,
+it is expected to contribute $2N/(k-1)$ to the total tree length.
+By summing over $n < k \le N$,
+the $N$ tips of a tree are expected to descend from only $n$ lineages
+around $2(N/n - 1)$ generations ago.
+Inverting this relationship between time and number of roots
+implies that a marginal tree cut $T$ units of time ago
+% since 2N(1/n - 1/N) = 2(N/n-1) = T implies
+%    N/n = (T+2)/2 and so n = 2N/(T+2)
+is expected to have around $r(T)$ roots, where $r(T) \approx 2N/(T+2)$.
 The total tree length over this time is
-$2 N \sum_{k=n(T)+1}^{N-1} 1/k$, which
-% is approximately
+$2N \sum_{k=N}^{r(T)+1} 1/(k-1) \approx 2 N \log(N/r(T))$.
+This is a crude estimate for several reasons:
+first, we should not count the branch leading to the root of the tree (i.e., when $r(T)=1$),
+and second, this does not account for the discrete nature of the Wright--Fisher model.
+% which is approximately
 % \begin{align*}
-%     2 N \log\left( \frac{N}{1 + 2 N/(T+2)} \right) .
+%     2N - 2 + 8 N \left\{ \log(N-1) - \log(r(T)) \}
+%        = 2N - 2 + 8 N \log( (T+2)(N-1)/2N )
+%        \approx 2 N (1 + 4 \log( (T+2)/2 ))
 % \end{align*}
-leads to an upper bound on the number of edges of
+Nonetheless, this leads as above to an upper bound on the number of edges of
 \begin{align}
     \label{eqn:edge_bound}
-    2 N \left( 1 + 4 \log\left( \frac{NT}{T + 2 N} \right)\right) .
+    2 N \left( 1 + 4 \log\left( \min\left(N, \frac{T+2}{2}\right) \right)\right) .
```

Comment:
This makes sense based on the above, but isn't what we're plotting now.

Reply:
It's not?  Oh, right I [forgot to take out that `- 1/N`](https://github.com/petrelharp/ftprime_ms/blob/dc57882a131e038498ff360b4ef6b356e202db9d/sims/space-complexity.py#L68). Besides that, it agrees, no?

Reply:
Matches now, ta!